llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2966,7 +2966,15 @@ struct VPWidenLoadRecipe final : public VPWidenMemoryRecipe, public VPValue {
                                  getDebugLoc());
   }
 
-  VP_CLASSOF_IMPL(VPDef::VPWidenLoadSC);
+  static inline bool classof(const VPRecipeBase *R) {
```

Comment:
I'm not sure how this works. Isn't there already an actual VPWidenLoadEVLRecipe that corresponds to VPWidenLoadEVLSC?

Reply:
Yes. But we need to add the derived class's ID to the parent class `classof()` implementation. 
Otherwise `isa<VPWidenLoadRecipe>(VPWidenLoadEVLRecipe)` return `false`.

We can found that there are some similar implementation such as `VPReductionRecipe::classof()` and `VPWidenREcipe::classof()`.

Reply:
Oh ok, it's just both VPWidenLoadRecipe and VPWidenLoadEVLRecipe have a common parent so I was wondering why we can't just do: `isa<VPWidenMemoryRecipe>(VPWidenLoadEVLRecipe)`?

Reply:
Hmm, but `VPWidenLoadEVLRecipe` isn't a subclass of `VPWidenLoadRecipe`, so the `classof` here will be incorrect?

Reply:
Oops. My bad. I will close this PR.